tsconfig.base.json

```diff
@@ -0,0 +1,23 @@
+{
+  "include": [
+    "src/**/*",
+    "types/**/*"
+  ],
+  "compilerOptions": {
+    "moduleResolution": "node",
+    "declaration": true,
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "resolveJsonModule": true,
+    "strict": true,
```

Comment:
Just add a few more configs like the one in tracking SDK, fix the violations, and we should be good to go
```json
    "forceConsistentCasingInFileNames": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
```

Reply:
These rules are added in now.